### PR TITLE
Extract some types exported from async-loaded that are referenced in sync-loaded modules into their own files.

### PR DESCRIPTION
**Pull request by iclanton**
This is required by https://github.com/microsoft/rushstack/pull/3840

### CODE REVIEW

File: libraries/rush-lib/src/logic/PackageJsonUpdater.ts [hunk 1]
```diff
@@ -6,7 +6,8 @@ import * as semver from 'semver';
 import { ConsoleTerminalProvider, Terminal, ITerminalProvider, Colors } from '@rushstack/node-core-library';
 
 import { RushConfiguration } from '../api/RushConfiguration';
-import { BaseInstallManager, IInstallManagerOptions } from './base/BaseInstallManager';
+import { BaseInstallManager } from './base/BaseInstallManager';
+import { IInstallManagerOptions } from './base/BaseInstallManagerTypes';
```
Comment (iclanton) [inline libraries/rush-lib/src/logic/PackageJsonUpdater.ts:10]:
```suggestion
import type { IInstallManagerOptions } from './base/BaseInstallManagerTypes';
```

File: libraries/rush-lib/src/logic/PackageJsonUpdater.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@ import * as semver from 'semver';
 import { ConsoleTerminalProvider, Terminal, ITerminalProvider, Colors } from '@rushstack/node-core-library';
 
 import { RushConfiguration } from '../api/RushConfiguration';
-import { BaseInstallManager, IInstallManagerOptions } from './base/BaseInstallManager';
+import { BaseInstallManager } from './base/BaseInstallManager';
```
Comment (iclanton) [inline libraries/rush-lib/src/logic/PackageJsonUpdater.ts:9]:
```suggestion
import type { BaseInstallManager } from './base/BaseInstallManager';
```

File: libraries/rush-lib/src/logic/installManager/RushInstallManager.ts [hunk 3]
```diff
@@ -18,7 +18,8 @@ import {
 } from '@rushstack/node-core-library';
 import { PrintUtilities } from '@rushstack/terminal';
 
-import { BaseInstallManager, IInstallManagerOptions } from '../base/BaseInstallManager';
+import { BaseInstallManager } from '../base/BaseInstallManager';
```
Comment (apostolisms) [inline libraries/rush-lib/src/logic/installManager/RushInstallManager.ts:21]: type ?

**Review by apostolisms — APPROVED** [review]

File: libraries/rush-lib/src/logic/installManager/RushInstallManager.ts (hunk 3 above)
Comment (iclanton) [reply libraries/rush-lib/src/logic/installManager/RushInstallManager.ts:21]: This is used in an `extends`.

File: libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts [hunk 4]
```diff
@@ -6,7 +6,8 @@ import * as path from 'path';
 import * as semver from 'semver';
 import { FileSystem, FileConstants, AlreadyReportedError, Async } from '@rushstack/node-core-library';
 
-import { BaseInstallManager, IInstallManagerOptions } from '../base/BaseInstallManager';
+import { BaseInstallManager } from '../base/BaseInstallManager';
```
[the remark this answers is not in the document]
Comment (iclanton) [reply libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts:9]: This is used in an `extends`.

**Pull request merged**